### PR TITLE
Min consts as liability params

### PR DESCRIPTION
**Pull request by maneva3**

### CODE REVIEW

File: packages/currency/src/lpn.rs [hunk 1]
```diff
@@ -25,8 +25,7 @@ define_symbol! {
 }
 define_currency!(Usdc, USDC);
 
-#[derive(Clone, PartialEq, Eq, JsonSchema, Serialize, Deserialize)]
-#[cfg_attr(any(test, feature = "testing"), derive(Debug))]
+#[derive(Clone, PartialEq, Eq, JsonSchema, Serialize, Deserialize, Debug)]
```
[the remark this answers is not in the document]
Comment (maneva3) [reply packages/currency/src/lpn.rs:28]: this change was made due to the error: "Lpns cannot be formatted using `{:?}` ..". The group is used in packages/finance/src/liability/dto/mod.rs line 37

File: packages/finance/src/liability/dto/mod.rs [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+    /// The percentage above which the second liquidity warning is issued.
+    second_liq_warn: Percent,
+    /// The percentage above which the third liquidity warning is issued.
+    third_liq_warn: Percent,
+    /// The maximum percentage of the amount due versus the locked collateral
+    /// max > healthy
+    max: Percent,
+    /// The minimum overdue amount that could be collected
+    min_liq_amount: LpnCoin,
+    //  The minimum amount a lease should have before being liquidated fully
```
[the remark this answers is not in the document]
Comment (gmanev7) [reply packages/finance/src/liability/dto/mod.rs]: // is not a valid rustdoc delimiter, please use ///

File: packages/finance/src/liability/dto/mod.rs [hunk 3]
```diff
@@ -120,21 +120,8 @@ where
...
-
-    fn try_from(dto: &LiabilityDTO) -> Result<Self> {
-        let liability_dto = dto.clone();
-
-        let min_liq = liability_dto.min_liquidation.try_into()?;
-        let min_asset = liability_dto.min_asset.try_into()?;
+        let min_liq = dto.min_liquidation.try_into()?;
+        let min_asset = dto.min_asset.try_into()?;
 
         let min_liq_asset_result = (min_liq, min_asset);
```
Comment (gmanev7) [inline packages/finance/src/liability/dto/mod.rs:126]: why do you compose them into a pair?

File: packages/finance/src/liability/dto/mod.rs [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+use super::Liability;
+
+mod unchecked;
+
+pub type LpnCoin = CoinDTO<Lpns>;
+
+#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+#[serde(try_from = "unchecked::LiabilityDTO")]
+pub struct LiabilityDTO {
```
Comment (gmanev7) [inline packages/finance/src/liability/dto/mod.rs:22]: Please make sure all attributes match the one of the previous `struct Liability`

**Review by gmanev7 — COMMENTED** [review]
to be continued ...

File: contracts/lease/src/contract/state/opening/request_loan.rs [hunk 5]
```diff
@@ -35,18 +35,19 @@ impl RequestLoan {
...
 
         let OpenLoanReqResult { batch, downpayment } = lpp.clone().execute_lender(
             OpenLoanReq::new(
-                &spec.form.liability,
+                lease_form.liability,
                 info.funds,
-                spec.form.max_ltd,
+                lease_form.max_ltd,
                 oracle.clone(),
                 &deps.querier,
```
Comment (gmanev7) [inline contracts/lease/src/contract/state/opening/request_loan.rs:52]: please revert and provide `impl<Lpn> TryFrom<&LiabilityDTO> for Liability<Lpn>` instead

**Pull request closed without merging**